libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_PRO/README.md

```diff
@@ -0,0 +1,99 @@
+# TBS LUCID PRO Flight Controller
+
+The TBS LUCID PRO/Freestyle is a flight controller produced by [Team Black Sheep](https://www.team-blacksheep.com/).
+
+## Features
+
+ - MCU - STM32F405 32-bit processor running at 168 MHz
+ - IMU - ICM42688 (Freestyle) or MPU6000 (Pro)
+ - Barometer - SPL06
+ - OSD - AT7456E
+ - Onboard Flash: 4MByte
+ - 6x UARTs
+ - 6x PWM Outputs (4 Motor Output, 1 Camera PWM and 1 LED)
+ - Battery input voltage: 2S-6S
+ - BEC 3.3V 0.5A
+ - BEC 5V 3A
+ - BEC 9V 3A for video
+
+## Pinout
+
+![TBS LUCID PRO/Freestyle Board](TopBottom.png "TBS LUCID PRO/Freestyle")
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (RX, DMA-enabled)
+ - SERIAL2 -> UART2 (ESC Telem)
+ - SERIAL3 -> UART3 (HD RX, DMA-Enabled)
+ - SERIAL4 -> UART4 (SmartAudio)
+ - SERIAL5 -> UART5 (SBUS RX)
+ - SERIAL6 -> UART6 (GPS, DMA-Enabled)
+
+## RC Input
+
+RC input is configured by default via SERIAL1. It supports all serial RC protocols except PPM and SBUS. For SBUS
+use SERIAL5.
+
+Note: If the receiver is FPort the receiver must be tied to the SERIAL1 TX pin , RSSI_TYPE set to 3,
+and SERIAL1_OPTIONS must be set to 7 (invert TX/RX, half duplex). For full duplex like CRSF/ELRS use both
+RX1 and TX1 and set RSSI_TYPE also to 3.
+ 
+## FrSky Telemetry
+ 
+FrSky Telemetry is supported using an unused UART, such as the T3 pin (UART3 transmit).
+You need to set the following parameters to enable support for FrSky S.PORT:
+ 
+  - SERIAL3_PROTOCOL 10
+  - SERIAL3_OPTIONS 7
+  
+## OSD Support
+
+The TBS LUCID PRO supports OSD using OSD_TYPE 1 (MAX7456 driver) and simultaneously DisplayPort using TX3/RX3 on the HD VTX connector.
+
+## PWM Output
+
+The TBS LUCID PRO supports up to 6 PWM or DShot outputs. The pads for motor output
+M1 to M4 are provided on both the motor connector and on separate pads, plus
+M5-6 on a separate pads for LED strip and camera control.
+
+The PWM is in 3 groups:
+
+ - PWM 1-2   in group1
```

Comment:
@andyp1per does this work for BDShot? this timer shares DMA with SPI

Reply:
Yes it works